src/ffi/mod.rs

```diff
@@ -831,6 +844,72 @@ extern "C" {
     fn fortify_source();
 }
 
+#[cfg(feature = "hide_ips")]
+struct LogCensor {
+    ip_mask_seed: [u8; 32],
+    ip_masks: parking_lot::Mutex<HashMap<Ipv4Addr, String>>,
+    ip_regex: Regex,
+}
+
+#[cfg(feature = "hide_ips")]
+impl LogCensor {
+    fn new() -> Option<Self> {
+        Regex::new(
+            r"(?x)\b
+                (?:
+                    (?:
+                        25[0-5]
+                        |  2[0-4][0-9]
+                        |  1[0-9]{2}
+                        |  [1-9]?[0-9]
+                    )
+                    \.
+                ){3}
+                (?:
+                    25[0-5]
+                    |  2[0-4][0-9]
+                    |  1[0-9]{2}
+                    |  [1-9]?[0-9]
+                )
+                \b",
+        )
+        .ok()
+        .map(|re| LogCensor {
+            ip_masks: parking_lot::Mutex::new(HashMap::new()),
+            ip_mask_seed: rand::thread_rng().gen::<[u8; 32]>(),
+            ip_regex: re,
+        })
+    }
+
+    fn censor_logs(&self, log: String) -> String {
+        // Gather all of the IPs
+        self.ip_regex
+            .replace_all(&log, |captures: &Captures| {
+                // First capture is guaranteed to be present by regex crate
+                #[allow(index_access_check)]
+                let ip_match = &captures[0];
+                Ipv4Addr::from_str(ip_match)
+                    .map(|ip_addr| {
+                        self.ip_masks
+                            .lock()
+                            .entry(ip_addr)
+                            .or_insert_with(|| {
+                                // Probably good enough protection for this usecase
+                                let mut hasher = Sha256::new();
```

Comment:
**IF** we are using cache (which I don't think is a good idea) and compute a replacement string only once for a given sequence, than we don't need to use any deterministic function and we could just generated random replacement string.

Reply:
I'll get rid of the hashmap